app/Utilities/AdsUtils.php

```diff
@@ -20,8 +20,47 @@
 
 namespace Adshares\Adserver\Utilities;
 
+use Adshares\Ads\Util\AdsValidator;
+
 final class AdsUtils
 {
+    /**
+     * Minimum transfer fee `TXS_MIN_FEE`
+     */
+    const TXS_MIN_FEE = 10000;
+    /**
+     * Local transfer coefficient `TXS_PUT_FEE`
+     */
+    const TXS_LOCAL_FEE = 0.0005;
+    /**
+     * Remote transfer coefficient `TXS_LNG_FEE`
+     */
+    const TXS_REMOTE_FEE = 0.0005;
+
+    /**
+     * Calculates transfer fee.
+     * @param string $addressFrom sender address
+     * @param string $addressTo recipient address
+     * @param int $amount amount
+     * @return int transfer fee or -1, if invalid data provided
+     */
+    public static function calculateFee(string $addressFrom, string $addressTo, int $amount)
```

Comment:
I'd make the return type `int` and also, validation should not be the resposibility of this method or at least it should throw an exception.

Reply:
Validation was moved to parent function.